include/bout/index_derivs_interface.hxx

```diff
@@ -96,6 +96,9 @@ T flowDerivative(const T& vel, const T& f, CELL_LOC outloc, const std::string& m
   T result(localmesh);
   result.allocate(); // Make sure data allocated
   result.setLocation(outloc);
+  if (std::is_base_of<Field3D, T>::value) {
```

Comment:
It would be good to avoid special handling for particular field types so would it be possible to ensure all field types can provide this information? (also in the long run we might need to be able to identify if a Field2D is compatible with a Field3D etc.)

Reply:
I'm not sure what the right design would be. Things to consider:
* `Field2D` and `Field3D` probably need different labelling - `Field2D` is the same for `Orthogonal` and `FieldAligned` coordinate systems
  * could have different `COORDINATE_SYSTEM` enum classes for `Field3D` and `Field2D` plus some methods/functions to compare them
  * or extend `COORDINATE_SYSTEM` to something like `enum class COORDINATE_SYSTEM = {None, FieldAligned, Orthogonal, FieldAligned2D, FCI}` where `FieldAligned2D` would be used for `Field2D` and compatible with both `FieldAligned` and `Orthogonal` (suggestions for better naming welcome if we want to go with a scheme like this)
* In the final implementation, `Field2D` members of `Coordinates` need `coordinate_system` setting. Best thing to do might be split initialization with 'uninitialized' `ParallelTransform` objects constructed early on (so we can `getCoordinateSystem()` from them), before most of the initialization of `Coordinates`.

Reply:
`Field2D` could have `COORDINATE_SYSTEM::AxisSymmetric`?

Implementing `operator==(COORDINATE_SYSTEM, COORDINATE_SYSTEM)` would give us control over comparisons.